### PR TITLE
Changelog v2.9.0-beta.1

### PR DESCRIPTION
**Pull request by karimnaaji**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,21 +1,43 @@
+## 2.9.0-beta.1
+
+### Features ✨
+
+* Add `globe` projection. This new projection allow the design of 3d globe and can be enabled by either passing `projection: globe` to the map constructor or by calling `map.setProjection('globe')`. All layers are supported by globe except for Custom Layers and Sky.
```
Comment (SnailBones) [inline CHANGELOG.md:5]:
```suggestion
* Add `globe` projection. This new projection displays the map as a 3d globe and can be enabled by either passing `projection: globe` to the map constructor or by calling `map.setProjection('globe')`. All layers are supported by globe except for Custom Layers and Sky.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,21 +1,43 @@
...
+* Extend atmospheric `fog` with three new style specification properties: `high-color`, `space-color` and `star-intensity` to allow the design of atmosphere around the globe. ([#11590](https://github.com/mapbox/mapbox-gl-js/pull/11590))
```
Comment (SnailBones) [inline CHANGELOG.md:6]: Stars can also be used as a new feature to create night skies outside of globe, no? Maybe we should mention this here (also something to keep in mind for docs examples).

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+* Add a new line layer paint property in the style specification: `{"line-trim-offset", [trim-start, trim-end]}`, the property will only be effective when `line-gradient` property is set. The line part between `[trim-start, trim-end]` will be marked as transparent to make a line gradient a vanishing effect. The line trim-off percentage is based on the whole line range `[0.0, 1.0]`. If either `trim-start` or `trim-end` offset is out of valid range, the default range `[0.0, 0.0]` will be set. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```
Comment (SnailBones) [inline CHANGELOG.md:7]: This feels very verbose, can we cut it down and direct readers to docs or the PR for specific instructions?

Comment (karimnaaji) [reply CHANGELOG.md:7]: Yep, I already trimmed it a bit but it's still too much, I'll shorten again.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,21 +1,43 @@
...
+* Add an option for providing a geolocation adapter to `GeolocateControl` ([#11614](https://github.com/mapbox/mapbox-gl-js/pull/11614))
```
Comment (SnailBones) [inline CHANGELOG.md:8]: Hat tip! Also IMO liking to the original PR is more useful.
```suggestion
* Add an option for providing a geolocation adapter to `GeolocateControl` ([#10400](https://github.com/mapbox/mapbox-gl-js/pull/10400)) (h/t [behnammodi](https://github.com/behnammodi))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,21 +1,43 @@
...
+* Extend atmospheric `fog` with three new style specification properties: `high-color`, `space-color` and `star-intensity` to allow the design of atmosphere around the globe. ([#11590](https://github.com/mapbox/mapbox-gl-js/pull/11590))
+* Add a new line layer paint property in the style specification: `{"line-trim-offset", [trim-start, trim-end]}`, the property will only be effective when `line-gradient` property is set. The line part between `[trim-start, trim-end]` will be marked as transparent to make a line gradient a vanishing effect. The line trim-off percentage is based on the whole line range `[0.0, 1.0]`. If either `trim-start` or `trim-end` offset is out of valid range, the default range `[0.0, 0.0]` will be set. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
+* Add an option for providing a geolocation adapter to `GeolocateControl` ([#10400](https://github.com/mapbox/mapbox-gl-js/pull/10400)) (h/t [behnammodi](https://github.com/behnammodi))
```
Comment (SnailBones) [inline CHANGELOG.md:8]:
```suggestion
* Add an option for providing a geolocation adapter to `GeolocateControl`. ([#10400](https://github.com/mapbox/mapbox-gl-js/pull/10400)) (h/t [behnammodi](https://github.com/behnammodi))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,21 +1,43 @@
...
+* Add `Map.Title` property to locale options to localise the map `aria-label`. ([#11549](https://github.com/mapbox/mapbox-gl-js/pull/11549)) (h/t [andrewharvey](https://github.com/andrewharvey))
+* Allow duplicated coordinates in tile request URLs. ([#11441](https://github.com/mapbox/mapbox-gl-js/pull/11441)) (h/t [ozero](https://github.com/ozero))
+* Add `source` field requirement to terrain exaggeration in the style specification.  ([#11664](https://github.com/mapbox/mapbox-gl-js/pull/11664))
```
Comment (SnailBones) [inline CHANGELOG.md:11]: I don't think this is a feature. Maybe this should be in Bug Fixes of the style spec changelog?
```suggestion
```

File: src/style-spec/CHANGELOG.md [hunk 7]
```diff
@@ -1,11 +1,21 @@
+## 13.25.0-beta.1
+
+### Features ✨
+
+* Extend atmospheric `fog` with three new style specification properties: `high-color`, `space-color` and `star-intensity` to allow the design of atmosphere around the globe ([#11590](https://github.com/mapbox/mapbox-gl-js/pull/11590))
+* Add a new line layer paint property in the style specification: `{"line-trim-offset", [trim-start, trim-end]}`, the property will only be effective when `line-gradient` property is set. The line part between `[trim-start, trim-end]` will be marked as transparent to make a line gradient a vanishing effect. The line trim-off percentage is based on the whole line range `[0.0, 1.0]`. If either `trim-start` or `trim-end` offset is out of valid range, the default range `[0.0, 0.0]` will be set. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```
Comment (SnailBones) [inline src/style-spec/CHANGELOG.md]: Same as in the main changelog, this is very long.

File: src/style-spec/CHANGELOG.md [hunk 8]
```diff
@@ -1,11 +1,21 @@
...
+* Add `source` field requirement to terrain exaggeration in the style specification.  ([#11664](https://github.com/mapbox/mapbox-gl-js/pull/11664))
```
Comment (SnailBones) [inline src/style-spec/CHANGELOG.md:10]: Is this a bugfix?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,21 +1,43 @@
...
 
-## Performance improvements 🏁
+### Performance improvements 🏁
+
 * Improve memory usage by freeing memory more eagerly after loading tiles. ([#11434](https://github.com/mapbox/mapbox-gl-js/pull/11434))
 * Improve memory usage by reducing repeated line labels on overscaled tiles. ([#11414](https://github.com/mapbox/mapbox-gl-js/pull/11414))
 * Improve performance when placing many symbols on terrain. ([#11466](https://github.com/mapbox/mapbox-gl-js/pull/11466))
 
 ### Bug fixes 🐞
+
```
Comment (SnailBones) [inline CHANGELOG.md:39]: Nice formatting catches

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,21 +1,42 @@
+## 2.9.0-beta.1
+
+### Features ✨
+
+* Add `globe` projection. This new projection displays the map as a 3d globe and can be enabled by either passing `projection: globe` to the map constructor or by calling `map.setProjection('globe')`. All layers are supported by globe except for Custom Layers and Sky.
+* Extend atmospheric `fog` with three new style specification properties: `high-color`, `space-color` and `star-intensity` to allow the design of atmosphere around the globe and night skies. ([#11590](https://github.com/mapbox/mapbox-gl-js/pull/11590))
+* Add a new line layer paint property in the style specification: `{"line-trim-offset", [trim-start, trim-end]}`, the property will only be effective when `line-gradient` property is set. The line part between `[trim-start, trim-end]` will be marked as transparent to make a vanishing effect. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```
Comment (SnailBones) [inline CHANGELOG.md:7]: Nice! Can we get this even shorter?
```suggestion
* Add a new line layer paint property in the style specification: `line-trim-offset` that can be used to create a custom fade out with improved update performance over `line-gradient`. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```

File: src/style-spec/CHANGELOG.md [hunk 11]
```diff
@@ -1,11 +1,24 @@
+## 13.25.0-beta.1
+
+### Features ✨
+
+* Extend atmospheric `fog` with three new style specification properties: `high-color`, `space-color` and `star-intensity` to allow the design of atmosphere around the globe and night skies. ([#11590](https://github.com/mapbox/mapbox-gl-js/pull/11590))
+* Add a new line layer paint property in the style specification: `{"line-trim-offset", [trim-start, trim-end]}`, the property will only be effective when `line-gradient` property is set. The line part between `[trim-start, trim-end]` will be marked as transparent to make a vanishing effect. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```
Comment (karimnaaji) [inline src/style-spec/CHANGELOG.md:6]:
```suggestion
* Add a new line layer paint property in the style specification: `line-trim-offset` that can be used to create a custom fade out with improved update performance over `line-gradient`. ([#11570](https://github.com/mapbox/mapbox-gl-js/pull/11570))
```

**Review by SnailBones — APPROVED** [review]
👍

**Pull request merged**